tests/xharness/Jenkins/TestSelector.cs

```diff
@@ -202,67 +209,50 @@ void SelectTestsByLabel (int pullRequest)
 
 			MainLog.WriteLine ($"In total found {labels.Count ()} label(s): {string.Join (", ", labels.ToArray ())}");
 
-			// disabled by default
-			SetEnabled (labels, "mtouch", ref jenkins.IncludeMtouch);
-			SetEnabled (labels, "mmp", ref jenkins.IncludeMmpTest);
-			SetEnabled (labels, "bcl", ref jenkins.IncludeBcl);
-			SetEnabled (labels, "bcl-xunit", ref jenkins.IncludeBCLxUnit);
-			SetEnabled (labels, "bcl-nunit", ref jenkins.IncludeBCLNUnit);
-			SetEnabled (labels, "mscorlib", ref jenkins.IncludeMscorlib);
-			SetEnabled (labels, "btouch", ref jenkins.IncludeBtouch);
-			SetEnabled (labels, "mac-binding-project", ref jenkins.IncludeMacBindingProject);
-			SetEnabled (labels, "ios-extensions", ref jenkins.IncludeiOSExtensions);
-			SetEnabled (labels, "device", ref jenkins.IncludeDevice);
-			SetEnabled (labels, "xtro", ref jenkins.IncludeXtro);
-			SetEnabled (labels, "cecil", ref jenkins.IncludeCecil);
-			SetEnabled (labels, "old-simulator", ref jenkins.IncludeOldSimulatorTests);
-			SetEnabled (labels, "all", ref jenkins.IncludeAll);
-
-			// enabled by default
-			SetEnabled (labels, "ios-32", ref jenkins.IncludeiOS32);
-			SetEnabled (labels, "ios-64", ref jenkins.IncludeiOS64);
-			SetEnabled (labels, "ios", ref jenkins.IncludeiOS); // Needs to be set after `ios-32` and `ios-64` (because it can reset them)
-			SetEnabled (labels, "tvos", ref jenkins.IncludetvOS);
-			SetEnabled (labels, "watchos", ref jenkins.IncludewatchOS);
-			SetEnabled (labels, "mac", ref jenkins.IncludeMac);
-			SetEnabled (labels, "ios-msbuild", ref jenkins.IncludeiOSMSBuild);
-			SetEnabled (labels, "ios-simulator", ref jenkins.IncludeSimulator);
-			SetEnabled (labels, "non-monotouch", ref jenkins.IncludeNonMonotouch);
-			SetEnabled (labels, "monotouch", ref jenkins.IncludeMonotouch);
+			foreach (var keyValuePair in testNameSelectionMatch) {
+				SetEnabled (labels, keyValuePair.Key, keyValuePair.Value);
+			}
 
-			bool inc_permission_tests = false;
-			if (SetEnabled (labels, "system-permission", ref inc_permission_tests))
-				Harness.IncludeSystemPermissionTests = inc_permission_tests;
+			// special case since we are not working with the TestSelection flag in jenkins
+			Harness.IncludeSystemPermissionTests = labels.Contains ("run-system-permission-tests") || jenkins.TestSelection.HasFlag (TestSelection.All);
 
 			// docs is a bit special:
 			// - can only be executed if the Xamarin-specific parts of the build is enabled
 			// - enabled by default if the current branch is master (or, for a pull request, if the target branch is master)
-			var changed = SetEnabled (labels, "docs", ref jenkins.IncludeDocs);
+			var changed = SetEnabled (labels, "docs", TestSelection.Docs);
```

Comment:
`SetEnabled` doesn't return `true` in this case for `skip-all-tests` anymore, which means we will run the documentation tests if the target branch for a PR is `master` even when `skip-all-tests` is set.